enterprise/coderd/aibridge_test.go

```diff
@@ -0,0 +1,456 @@
+					StartedAfter: i2.StartedAt.Add(-10 * time.Minute),
+				},
+				want: []codersdk.AIBridgeInterception{i1SDK, i2SDK},
+			},
+			{
+				name: "StartedBefore/NoMatch",
+				filter: codersdk.AIBridgeListInterceptionsFilter{
+					StartedBefore: i3.StartedAt.Add(-10 * time.Minute),
+				},
+				want: []codersdk.AIBridgeInterception{},
+			},
+			{
+				name: "StartedBefore/OK",
+				filter: codersdk.AIBridgeListInterceptionsFilter{
+					StartedBefore: i3.StartedAt.Add(10 * time.Minute),
+				},
+				want: []codersdk.AIBridgeInterception{i3SDK},
+			},
+			{
+				name: "BothBeforeAndAfter/NoMatch",
+				filter: codersdk.AIBridgeListInterceptionsFilter{
+					StartedAfter:  i1.StartedAt.Add(10 * time.Minute),
+					StartedBefore: i1.StartedAt.Add(20 * time.Minute),
+				},
+				want: []codersdk.AIBridgeInterception{},
+			},
+			{
+				name: "BothBeforeAndAfter/OK",
+				filter: codersdk.AIBridgeListInterceptionsFilter{
+					StartedAfter:  i2.StartedAt.Add(-10 * time.Minute),
+					StartedBefore: i2.StartedAt.Add(10 * time.Minute),
+				},
+				want: []codersdk.AIBridgeInterception{i2SDK},
+			},
+		}
+
+		for _, tc := range cases {
+			t.Run(tc.name, func(t *testing.T) {
+				t.Parallel()
+				ctx := testutil.Context(t, testutil.WaitLong)
+				res, err := experimentalClient.AIBridgeListInterceptions(ctx, tc.filter)
+				require.NoError(t, err)
+				// We just compare UUID strings for the sake of this test.
+				wantIDs := make([]string, len(tc.want))
+				for i, r := range tc.want {
+					wantIDs[i] = r.ID.String()
+				}
+				gotIDs := make([]string, len(res.Results))
+				for i, r := range res.Results {
+					gotIDs[i] = r.ID.String()
+				}
+				require.Equal(t, wantIDs, gotIDs)
+			})
+		}
+	})
+
+	t.Run("FilterErrors", func(t *testing.T) {
+		t.Parallel()
+		dv := coderdtest.DeploymentValues(t)
+		dv.Experiments = []string{string(codersdk.ExperimentAIBridge)}
+		client, _ := coderdenttest.New(t, &coderdenttest.Options{
+			Options: &coderdtest.Options{
+				DeploymentValues: dv,
+			},
+		})
+		experimentalClient := codersdk.NewExperimentalClient(client)
+
+		// No need to insert any test data, we're just testing the filter
+		// errors.
+
+		cases := []struct {
+			name string
+			q    string
+			want []codersdk.ValidationError
+		}{
+			{
+				name: "UnknownUsername",
+				q:    "initiator:unknown",
+				want: []codersdk.ValidationError{
+					{
+						Field:  "initiator",
+						Detail: `Query param "initiator" has invalid value: user "unknown" either does not exist, or you are unauthorized to view them`,
+					},
+				},
+			},
+			{
+				name: "InvalidStartedAfter",
+				q:    "started_after:invalid",
+				want: []codersdk.ValidationError{
+					{
+						Field:  "started_after",
+						Detail: `Query param "started_after" must be a valid date format (2006-01-02T15:04:05.999999999Z07:00): parsing time "INVALID" as "2006-01-02T15:04:05.999999999Z07:00": cannot parse "INVALID" as "2006"`,
```

Comment:
I worry about returning the Go time format here because it can be very confusing
i.e. "wtf is it talking abotu 2006?"